mypy.ini

```diff
@@ -2,7 +2,8 @@
 allow_redefinition = True
 check_untyped_defs = True
 ignore_missing_imports = True
-incremental = True
+# Incremental mode causes false-negatives
+incremental = False
```

Comment:
We had this flag to support mypy in incremental mode as the first-class-citizen. It is important for:
- large projects that use `--incremental` to speed up their checks
- IDEs that use mypy plugins

I am not sure that removing it is a good idea.

Reply:
Got it, okay. Reverted this and added explicit mention in CONTRIBUTING.md

Reply:
Can you please post unexpected results here? What were they?

Reply:
The unexpected result was quite simple: I started from this regression test and confirmed that it fails. Then I attempted first fix (added `module_name`) and rerun test suite - this reg.test passed as expected and two others failed. Then I reverted everything and... got fully passing test suite, including reg.test - which is obviously wrong. Removing mypy cache helped to make reg. test fail again, so incremental mode is harmful during development. Maybe we should have different configs for CI and local development?